File: clojure/src/spoonacular_api/specs/get_product_information_200_response.clj
(ns spoonacular-api.specs.get-product-information-200-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [spoonacular-api.specs. :refer :all]
            [spoonacular-api.specs.get-product-information-200-response-ingredients-inner :refer :all]
            [spoonacular-api.specs.search-grocery-products-by-upc-200-response-nutrition :refer :all]
            [spoonacular-api.specs.search-grocery-products-by-upc-200-response-servings :refer :all]
            )
  (:import (java.io File)))


(def get-product-information-200-response-data
  {
   (ds/req :id) int?
   (ds/req :title) string?
   (ds/req :breadcrumbs) (s/coll-of string?)
   (ds/req :imageType) string?
   (ds/req :badges) (s/coll-of string?)
   (ds/req :importantBadges) (s/coll-of string?)
   (ds/req :ingredientCount) int?
   (ds/opt :generatedText) any-type-spec
   (ds/req :ingredientList) string?
   (ds/req :ingredients) (s/coll-of get-product-information-200-response-ingredients-inner-spec)
   (ds/req :likes) float?
   (ds/req :aisle) string?
   (ds/req :nutrition) search-grocery-products-by-upc-200-response-nutrition-spec
   (ds/req :price) float?
   (ds/req :servings) search-grocery-products-by-upc-200-response-servings-spec
   (ds/req :spoonacularScore) float?
   })

(def get-product-information-200-response-spec
  (ds/spec
    {:name ::get-product-information-200-response
     :spec get-product-information-200-response-data}))

File: scala/src/main/scala/org/openapitools/models/GetProductInformation200Response.scala
package org.openapitools.models

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import spoonacular._
import org.openapitools.models.AnyType
import org.openapitools.models.BigDecimal
import org.openapitools.models.GetProductInformation200ResponseIngredientsInner
import org.openapitools.models.SearchGroceryProductsByUPC200ResponseNutrition
import org.openapitools.models.SearchGroceryProductsByUPC200ResponseServings
import scala.collection.immutable.Seq

/**
 * 
 * @param id 
 * @param title 
 * @param breadcrumbs 
 * @param imageType 
 * @param badges 
 * @param importantBadges 
 * @param ingredientCount 
 * @param generatedText 
 * @param ingredientList 
 * @param ingredients 
 * @param likes 
 * @param aisle 
 * @param nutrition 
 * @param price 
 * @param servings 
 * @param spoonacularScore 
 */
case class GetProductInformation200Response(id: Int,
                title: String,
                breadcrumbs: Seq[String],
                imageType: String,
                badges: Seq[String],
                importantBadges: Seq[String],
                ingredientCount: Int,
                generatedText: Option[AnyType],
                ingredientList: String,
                ingredients: Seq[GetProductInformation200ResponseIngredientsInner],
                likes: BigDecimal,
                aisle: String,
                nutrition: SearchGroceryProductsByUPC200ResponseNutrition,
                price: BigDecimal,
                servings: SearchGroceryProductsByUPC200ResponseServings,
                spoonacularScore: BigDecimal
                )

object GetProductInformation200Response {
    /**
     * Creates the codec for converting GetProductInformation200Response from and to JSON.
     */
    implicit val decoder: Decoder[GetProductInformation200Response] = deriveDecoder
    implicit val encoder: ObjectEncoder[GetProductInformation200Response] = deriveEncoder
}
